src/python/qeqiskit/optimizer/optimizer.py

```diff
@@ -1,36 +1,52 @@
-from zquantum.core.history.recorder import recorder
+import numpy as np
+from zquantum.core.history.recorder import recorder as _recorder
 from zquantum.core.interfaces.optimizer import (
     Optimizer,
     optimization_result,
     construct_history_info,
 )
+from zquantum.core.interfaces.functions import CallableWithGradient
+from zquantum.core.typing import RecorderFactory
+from typing import Optional, Dict
 from qiskit.algorithms.optimizers import SPSA, ADAM
 
 
 class QiskitOptimizer(Optimizer):
-    def __init__(self, method, options={}):
+    def __init__(
+        self,
+        method: str,
+        options: Optional[Dict] = None,
```

Comment:
i think `options` should be renamed to `optimizer_kwargs`.

Reply:
Additionally, it would make more sense to initialize `self.optimizer` in the `__init__`, hence moving lines 57-62 into the `__init__`.